server/sonar-main/src/main/java/org/sonar/application/command/CommandFactoryImpl.java

```diff
@@ -67,27 +67,72 @@ public CommandFactoryImpl(Props props, File tempDir, System2 system2) {
 
-    return new EsCommand(ProcessId.ELASTICSEARCH, esFileSystem.getHomeDirectory())
-      .setFileSystem(esFileSystem)
-      .setLog4j2Properties(new EsLogging().createProperties(props, esFileSystem.getLogDirectory()))
-      .setArguments(props.rawProperties())
+    elasticsearchConfiguration
+      .setLog4j2Properties(new EsLogging().createProperties(props, elasticsearchConfiguration.getLogDirectory()))
+      .setEsJvmOptions(new EsJvmOptions()
+        .addFromMandatoryProperty(props, ProcessProperties.SEARCH_JAVA_OPTS)
+        .addFromMandatoryProperty(props, ProcessProperties.SEARCH_JAVA_ADDITIONAL_OPTS))
+      .setEsYmlSettings(new EsYmlSettings(settingsMap))
       .setClusterName(settingsMap.get("cluster.name"))
       .setHost(settingsMap.get("network.host"))
-      .setPort(Integer.valueOf(settingsMap.get("transport.tcp.port")))
-      .addEsOption("-Epath.conf=" + esFileSystem.getConfDirectory().getAbsolutePath())
+      .setPort(Integer.valueOf(settingsMap.get("transport.tcp.port")));
+
+    return new ExternalCommand(ProcessId.ELASTICSEARCH, elasticsearchConfiguration.getHomeDirectory())
+      .setElasticsearchConfiguration(elasticsearchConfiguration)
+      .setArguments(props.rawProperties())
+      .addEsOption("-Epath.conf=" + elasticsearchConfiguration.getConfDirectory().getAbsolutePath())
+      .setEnvVariable("ES_JVM_OPTIONS", elasticsearchConfiguration.getJvmOptions().getAbsolutePath())
+      .setEnvVariable("JAVA_HOME", System.getProperties().getProperty("java.home"))
+      .suppressEnvVariable(ENV_VAR_JAVA_TOOL_OPTIONS);
+  }
+
+  private JavaCommand createEsCommandForWindows() {
+    ElasticsearchConfiguration elasticsearchConfiguration = new ElasticsearchConfiguration(props);
+    if (!elasticsearchConfiguration.getExecutable().exists()) {
+      throw new IllegalStateException("Cannot find elasticsearch binary");
+    }
+    Map<String, String> settingsMap = new EsSettings(props, elasticsearchConfiguration, System2.INSTANCE).build();
+
+    elasticsearchConfiguration
+      .setLog4j2Properties(new EsLogging().createProperties(props, elasticsearchConfiguration.getLogDirectory()))
       .setEsJvmOptions(new EsJvmOptions()
         .addFromMandatoryProperty(props, ProcessProperties.SEARCH_JAVA_OPTS)
         .addFromMandatoryProperty(props, ProcessProperties.SEARCH_JAVA_ADDITIONAL_OPTS))
       .setEsYmlSettings(new EsYmlSettings(settingsMap))
-      .setEnvVariable("ES_JVM_OPTIONS", esFileSystem.getJvmOptions().getAbsolutePath())
+      .setClusterName(settingsMap.get("cluster.name"))
+      .setHost(settingsMap.get("network.host"))
+      .setPort(Integer.valueOf(settingsMap.get("transport.tcp.port")));
+
+    return new JavaCommand<EsJvmOptions>(ProcessId.ELASTICSEARCH, elasticsearchConfiguration.getHomeDirectory())
+      .setElasticsearchConfiguration(elasticsearchConfiguration)
+      .setReadsArgumentsFromFile(false)
+      //.setArguments(props.rawProperties()) //FIXME set settings
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)